### PR TITLE
refactor: Add payload type checks

### PR DESCRIPTION
**Pull request by pawanpaudel93**
## Summary

This PR adds payload type checking to most of the contact functions.

### CODE REVIEW

File: src/pages/profile/components/Sidebar.tsx [hunk 1]
```diff
@@ -124,7 +124,7 @@ export default function Sidebar({
     Object.keys(updatedData).forEach((key: string) => {
       const typedKey = key as keyof User
 
-      if (updatedData[typedKey] && originalData[typedKey] !== updatedData[typedKey]) {
+      if (typeof updatedData[typedKey] === 'string' && originalData[typedKey] !== updatedData[typedKey]) {
```
Comment (kranthicodes) [inline src/pages/profile/components/Sidebar.tsx:127]: Can we use `isInvalidInput` function here and few other files?

File: warp/protocol-land/utils/isInvalidInput.ts [hunk 2]
```diff
@@ -1,15 +1,67 @@
...
  *                        Defaults to ['string'] if not provided.
+ * @param skipEmptyStringCheck - Whether to skip empty string checking. Defaults to false.
  * @returns True if the input is invalid otherwise false.
  */
-export const isInvalidInput = (input: any, expectedTypes: ExpectedType | ExpectedType[] = ['string']) =>
-  !input ||
-  !(Array.isArray(expectedTypes) ? expectedTypes : [expectedTypes]).some((expectedType) =>
-    expectedType === 'array' ? Array.isArray(input) : typeof input === expectedType
-  )
+export const isInvalidInput = (
```
Comment (kranthicodes) [inline warp/protocol-land/utils/isInvalidInput.ts:39]: Lets utilize this function as much as we can through out the code for consistency

File: src/pages/profile/components/Sidebar.tsx (hunk 1 above)
Comment (pawanpaudel93) [reply src/pages/profile/components/Sidebar.tsx:127]: Yes, we can use this function if we find suitable place to use it.

Here in this case simply checking if its a string will work for us as we are only updating the keys of user profile which are all string value at the moment.

File: warp/protocol-land/utils/isInvalidInput.ts (hunk 2 above)
Comment (pawanpaudel93) [reply warp/protocol-land/utils/isInvalidInput.ts:39]: Yes, i have used this function throughout the contract.

File: src/pages/profile/components/Sidebar.tsx (hunk 1 above)
Comment (7i7o) [reply src/pages/profile/components/Sidebar.tsx:127]: Agree with @kranthicodes here 😊 
It's easier to find all places where we are type checking if we re-use `isInvalidInput` everywhere

**Review by 7i7o — APPROVED** [review]
utACK

File: warp/protocol-land/actions/user.ts [hunk 3]
```diff
@@ -1,18 +1,62 @@
...
+    isInvalidInput(payload, 'object') ||
+    (payload.fullName !== undefined && isInvalidInput(payload.fullName, 'string', true)) ||
+    (payload.userName !== undefined && isInvalidInput(payload.userName, 'string', true)) ||
+    (payload.avatar !== undefined && isInvalidInput(payload.avatar, 'string', true)) ||
+    (payload.bio !== undefined && isInvalidInput(payload.bio, 'string', true)) ||
+    (payload.timezone !== undefined && isInvalidTimezone(payload.timezone)) ||
+    (payload.location !== undefined && isInvalidInput(payload.location, 'string', true)) ||
+    (payload.twitter !== undefined && isInvalidInput(payload.twitter, 'string', true)) ||
+    (payload.email !== undefined && isInvalidInput(payload.email, 'string', true)) ||
+    (payload.website !== undefined && isInvalidInput(payload.website, 'string', true)) ||
```
[the remark this answers is not in the document]
Comment (pawanpaudel93) [reply warp/protocol-land/actions/user.ts:33]: Yes we should. I will update it if i find some working good regexes.

[the remark this answers is not in the document]
Comment (pawanpaudel93) [reply warp/protocol-land/actions/user.ts]: email and url check has been added.

**Review by kranthicodes — APPROVED** [review]
utACK

**Pull request merged**